### ISSUE TITLE
Add a low-level setting that enables DoH3 negotation for all DoH upstreams

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForiOS#2183]
Currently AdGuard will use DoH2 exclusively for all DoH upstreams that use `https://` URLs.

We should add a low-level setting that enables both DoH3 and DoH2 (it's available as a flag in DnsLibs).

When this flag is enabled, DnsLibs tries to connect using both DoH3 and DoH2 simultaneously and chooses the one that was established faster, and then closes the slower one.

**Comment by FsAguard62** [comment]
I agree with this issue, here is what I get when I try to use the h3:// domain with the most resent update of AdGuard 
![31DCAF90-E628-4E81-9D38-A8F3A7FA2DD6](https://github.com/AdguardTeam/AdguardForiOS/assets/78290214/a9aa62ef-56d4-4b38-9e67-ac5e18fe1d8d)

**Comment by ameshkov** [comment]
@FsAguard62 several options:

1. DoH3 does not work at all (maybe QUIC is blocked?)
2. You're using native implementation

**Comment by FsAguard62** [comment]
Well, I have figured out with the help from Vladislav on the Adguard chat today it was the Adguard VPN stopping the connection, and you can't use the “h3://” domain in the AGVPN standalone service I get the same error in there![image](https://github.com/AdguardTeam/AdguardForiOS/assets/78290214/673d082c-9468-42f3-89a9-f35a97cd7576) too!

**Comment by FsAguard62** [comment]
I think it would be so much easier to have the selection of the service on the AdGuard (products) setting

**Comment by FsAguard62** [comment]
Just have another not related question about the domain “footprint” Why is the domain not being blocked by AdGuard it is a tracking site?
